pkg/apply/apply.go

```diff
@@ -103,8 +104,8 @@ func DeleteOwnedObject(ctx context.Context, client k8sclient.Client, obj *uns.Un
 	existing.SetGroupVersionKind(gvk)
 	err := client.Get(ctx, types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}, existing)
 	if err != nil {
-		if !apierrors.IsNotFound(err) {
-			return errors.Wrapf(err, "could not retrieve existing %s", objDesc)
+		if !apierrors.IsNotFound(err) && !apimeta.IsNoMatchError(err) {
```

Comment:
@qinqon could you explain what this does ?

Reply:
I will add a comment